### PR TITLE
cleanup superfluous dependencies in Snakemake 8

### PR DESCRIPTION
**Pull request by johanneskoester**
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/snakemake/meta.yaml [hunk 1]
```diff
@@ -1,41 +1,24 @@
...
-    # aws s3
-    - boto3
-    # check snakemake/setup.cfg's section [options.extras_require]
-    # options.extras_require: google-cloud
-    - google-cloud-storage
-    - google-api-python-client
-    - google-crc32c
-    - oauth2client
     # options.extras_require: messaging
     - slacker
```
Comment (corneliusroemer) [inline recipes/snakemake/meta.yaml:23]: should this be `slack_sdk`?

File: recipes/snakemake/meta.yaml [hunk 2]
```diff
@@ -93,13 +76,14 @@ outputs:
```
Comment (corneliusroemer) [inline recipes/snakemake/meta.yaml:76]: Might it be necessary to add change the entry points to:
```
  snakemake = snakemake.cli:main
  snakemake-bash-completion = snakemake.cli:bash_completion
```
?

**Review by corneliusroemer — COMMENTED** [review]
This is my attempt, mostly identical: https://github.com/bioconda/bioconda-recipes/pull/44923/files 

I've highlighted the two changes in setup.cfg that aren't reflected in this PR: `slack_sdk` and `snakemake.cli`

Setup.cfg diff for reference: https://github.com/snakemake/snakemake/compare/v7.32.4...v8.0.0#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52

**Comment by corneliusroemer** [comment]
Ci fails because:
```
The reported errors are:
19:45:27 BIOCONDA INFO (OUT) - Encountered problems while solving:
19:45:27 BIOCONDA INFO (OUT) -   - nothing provides requested conda-inject >=1.3.1,<2.0
19:45:27 BIOCONDA INFO (OUT) -   - nothing provides conda-inject >=1.3.1,<2.0 needed by snakemake-minimal-8.0.0-pyhdfd78af_0
```

https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=48023&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=4dddc55b-10bc-50fc-ac68-7ff899b7a030&l=1064

**Comment by johanneskoester** [comment]
Right, forgot to publish that on conda-forge. Give me a second.

**Comment by johanneskoester** [comment]
Done. Should be available in the CDN soon.

**Comment by corneliusroemer** [comment]
I've retriggered CI - can you look at my two review comments?

**Comment by johanneskoester** [comment]
conda-forge was still missing a newer version of conda-inject. Done that manually now: https://github.com/conda-forge/conda-inject-feedstock/pull/7

**Comment by corneliusroemer** [comment]
Checks pass! With bioconda not yet supporting Python 3.11 and snakemake requiring at least Python 3.11 this package will be nearly impossible to install until bioconda supports Python 3.11 I fear (unless one pip installs).

**Comment by johanneskoester** [comment]
> Checks pass! With bioconda not yet supporting Python 3.11 and snakemake requiring at least Python 3.11 this package will be nearly impossible to install until bioconda supports Python 3.11 I fear (unless one pip installs).
> 
> As long as _one_ non-noarch bioconda package is required things won't work.

Usually, one should not need to mix Snakemake with other packages in an environment, because of its conda/apptainer integration. However, maybe a good time to add python 3.11 as soon as possible (once the current bioconductor rebuild is over).

**Comment by johanneskoester** [comment]
(I have accidentlly merged the wrong PR, damn!). And now the linter complains about the build number being changed here for no reason.

**Comment by johanneskoester** [comment]
Closing in favor of #44934

**Pull request closed without merging**